src/_langs/en/updates/2015-07-22-mediastream-deprecations.markdown

```diff
@@ -0,0 +1,155 @@
+  streaming, not `MediaStream.ended`.
+* Use `MediaStreamTrack.stop()` to stop streaming, not `MediaStream.stop()`.
+* As ever, `MediaStream.id` provides a unique identifier for a `MediaStream`
+  (no change there) and `MediaStreamTrack.label` provides a human-readable
+  label for the source device for a stream, e.g. _FaceTime HD Camera (Built-in)
+  (05ac:8510)_.
+
+You can see these in action: open [simpl.info/gum](https://simpl.info/gum) in
+Chrome (on a device with a camera) and view the Chrome Dev Tools console. The
+MediaStream object `stream` passed to the `getUserMedia()` callback in this
+demo is in global scope, so you can inspect it from the console.  Call
+`stream.getTracks()[0]` to view the `MediaStreamTrack` for this stream.
+
+<p style="text-align: center;">
+  <img src="/web/updates/images/2015-07-22-mediastream-deprecations/mediastream-screenshot.png"
+  alt="Screenshot showing MediaStream and MediaStreamTrack in the Chrome Dev Tools console">
+</p>
+
+## Stop(), ended, active
+
+When the [W3C Working Group](http://w3c.github.io/mediacapture-main/getusermedia.html)
+looked at the problem of what happens when you add new tracks to a `MediaStream`,
+and whether an empty `MediaStream` is ended or not, they realized that there was
+no way to do `ended` (as in 'will never start again') on a `MediaStream` in
+a sensible way. In other parts of HTML5 'ended' means 'this has ended and will
+never resume'. 'Active' carries no such implication: an inactive stream can
+become active again, for instance if a new track is added to it. Rather than try
+to maintain a confusing attribute and function, the Working Group decided to
+remove it.
+
+Here's an example of how to use 'MediaStream.active' to check the status of a
+stream:
+
+    var gumStream;
+
+    navigator.getUserMedia({audio: false, video: true},
+        function(stream) {
+             gumStream = stream;
+            // ...
+        },
+        function(error) {
+            console.log('getUserMedia() error', error);
+        });
+
+    // …
+
+    if (gumStream.active) {
+        // do something with the stream
+    }
+
+Removing `stop()` from `MediaStream` did not remove any real functionality:
+processes for detaching source devices and so on have to be done on
+`MediaStreamTrack` anyway. Use `stop()` on `MediaStreamTrack` instead:
+
+    navigator.getUserMedia({audio: false, video: true},
+        function(stream) {
+             // can also use getAudioTracks() or getVideoTracks()
+            var track = stream.getTracks()[0];  // if only one media track
+            // ...
+            track.stop();
+        },
+        function(error){
+            console.log('getUserMedia() error', error);
+        });
+
+## label
+
+Turns out that nobody could quite figure out a use for this property!
+
+It had been added to the very first version of the spec, but nobody really knew
+what it was for, and it was completely unclear what happened to it when a stream
```

Comment:
what is was for, and it ...

Start a new sentence here.

what is was for. It was unclear what happened...

A few things to note:
* Yes, I ended with a preposition. It's a myth that you can't.
* 'Completely' seems unnecessary.